### PR TITLE
Lint all the things

### PR DESCRIPTION
**Pull request by kremalicious**
Basically change every js file to follow [ascribe's style guide](https://github.com/ascribe/javascript). Adds some missing dependencies too. Mostly removing semicolons as per https://github.com/ascribe/javascript/pull/46 to increase our hipness and shortening too long lines. Commented on the critical stuff in this PR though.

It's all perfectly consistent code from now on.

200w-1

Next step is to enforce code style by running it before every commit.

### CODE REVIEW

**Comment by kremalicious** [comment]
@diminator almost there, 2 to go, from like 30 initially. You wanna take a look at the remaining ones please?

screen shot 2017-06-12 at 17 26 13

File: src/connection/index.js [hunk 1]
```diff
@@ -33,22 +33,22 @@ export default class Connection {
...
      */
-    getStatus(tx_id) {
+    getStatus(txId) {
         return this._req(this.getApiUrls('statuses'), {
-                query: {
-                    tx_id
-                }
-            });
+            query: {
+                txId
```
Comment (kremalicious) [inline src/connection/index.js]: not so sure if renaming `tx_id` to `txId` is without consequences here. Can someone confirm if that's alright? More parameters renaming further down this file with `block_id` -> `blockId`

File: src/format_text.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import { sprintf } from 'sprintf-js';
+import { sprintf } from 'sprintf-js'
 
 
 // Regexes taken from or inspired by sprintf-js
 const Regex = {
-    TEMPLATE_LITERAL: /\${([^\)]+?)}/g,
+    TEMPLATE_LITERAL: /\${([^)]+?)}/g,
```
Comment (kremalicious) [inline src/format_text.js]: I suck at regex so need confirmation if removing the escaping here still produces same result

File: src/connection/index.js (hunk 1 above)
Comment (diminator) [inline src/connection/index.js]: Yeah, this would break the API:

good: `?tx_id=`
bad: `?txId=`

File: src/connection/index.js [hunk 3]
```diff
@@ -108,40 +108,40 @@ export default class Connection {
...
      */
-    listVotes(block_id) {
+    listVotes(blockId) {
         return this._req(this.getApiUrls('votes'), {
-                query: {
-                    block_id
-                }
-            });
+            query: {
+                blockId
```
Comment (diminator) [inline src/connection/index.js]: breaks the API

File: src/connection/index.js [hunk 4]
```diff
@@ -57,24 +57,24 @@ export default class Connection {
...
-    listBlocks({ tx_id, status }) {
+    listBlocks({ txId, status }) {
         return this._req(this.getApiUrls('blocks'), {
-                query: {
-                    tx_id,
-                    status
-                }
-            });
+            query: {
+                txId,
```
Comment (diminator) [inline src/connection/index.js]: breaks the API

File: src/format_text.js (hunk 2 above)
Comment (diminator) [inline src/format_text.js]: It should yield the same because of the surrounding square brackets `[ ]`

**Comment by diminator** [comment]
> @diminator almost there, 2 to go, from like 30 initially. You wanna take a look at the remaining ones please?

1. the `no-confusing-arrow` should be resolved by the curly brackets. can you confirm?
2. 
replace
```
            conditionJson.subfulfillments.forEach((subfulfillment) => {
                subfulfillment = ccJsonLoad(subfulfillment);
```
by

```
            conditionJson.subfulfillments.forEach((subfulfillmentJson) => {
                subfulfillment = ccJsonLoad(subfulfillmentJson);
```

**Comment by kremalicious** [comment]
awesome, all fixed. But the remaining `no-confusing-arrow` is weird.

current master:

```js
.then((res) => {
  return onlyJsonResponse ? res.json() :
  {
    json: res.json(),
    url: res.url
  };
  })
```

auto transformation by linting plugin in editor:

```js
.then((res) => onlyJsonResponse ? res.json() :
  {
    json: res.json(),
    url: res.url
})
```

if we do this, we produce another error (`no-unused-expressions`):

```js
.then((res) => {
  onlyJsonResponse ? res.json() :
  {
    json: res.json(),
    url: res.url
  }
})
```

Ideas? Otherwise I'm disabling the `no-confusing-arrow` for that line

**Pull request merged**